book/src/feature/lang_result.md

```diff
@@ -0,0 +1,30 @@
+# Result
+
+There is support for Rust's `std::Result`, where you can return either a value or an error.
+
+## Example
+
+```rust,noplayground
+
+pub enum CustomError {
```

Comment:
nit: Given that we support rich types of errors (this PR is good job!), such as nested types, struct types, etc, maybe we can document them here as well